src/harness/reference_models/interference/aggregate_interference.py

```diff
@@ -0,0 +1,366 @@
+#    Copyright 2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""
+==================================================================================
+  Aggregate Interference Calculations Reference Model[R2-SGN-12].
+  Based on WINNF-TS-0061, Version V1.0.0-r5.1, 17 January 2018.
+
+  This model provides the functions to calculate the aggregate interference for
+  the protection constrains of FSS-Cochannel, FSS-Blocking, ESC, GWPZ and PPA.
+
+  The main routines are:
+
+    calculateAggregateInterferenceForFssCochannel
+    calculateAggregateInterferenceForFssBlocking
+    calculateAggregateInterferenceForEsc
+    calculateAggregateInterferenceForGwpz
+    calculateAggregateInterferenceForPpa
+
+  The routines returns a nested dictionary in the format of
+  {latitude : {longitude : [interference1, interference2]}}. interference 
+  is the aggregate interference value in mW for each of the protection constraint.
+==================================================================================
+"""
+
+import numpy as np 
+import logging
+from reference_models.interference import interference as interf
+from reference_models.geo import utils
+import multiprocessing
+from multiprocessing import Pool
+from functools import partial
+
+
+# Aggregate interference to non-DPAs in the formation of 
+# {latitude: {longitude : [interference(mW), interference(mW)]}
+aggregate_interference = interf.AggregateInterferenceOutputFormat()
+
+
+def convertAndSumInterference(cbsd_interference_list):
+  """Converts interference in dBm to mW to calculate aggregate interference"""
+  
+  interferences_in_dbm = np.array(cbsd_interference_list)
+  return np.sum(interf.dbToLinear(interferences_in_dbm))
+
+
+def aggregateInterferenceForPoint(protection_point, channels, low_freq, high_freq, 
```

Comment:
This routine and the convertAndSumInterference really belongs closer from the other interference routines in `interference.py` file.
Can you move that on the file, which will be easy after your `git rebase` is done.

Reply:
Moved   to interference.py file